api/src/pcapi/core/finance/models.py

```diff
@@ -811,3 +813,96 @@ class PaymentMessage(Base, Model):
     id: int = sqla.Column(sqla.BigInteger, primary_key=True, autoincrement=True)
     name: str = sqla.Column(sqla.String(50), unique=True, nullable=False)
     checksum: bytes = sqla.Column(sqla.LargeBinary(32), unique=True, nullable=False)
+
+
+##
+# When an incident is `CREATED` it can be either `CANCELLED` and it's the end of this incident. Or it can be `VALIDATED`.
+##
+class IncidentStatus(enum.Enum):
+    CREATED = "created"
+    VALIDATED = "validated"
+    CANCELLED = "cancelled"
+
+
+class IncidentType(enum.Enum):
+    OVERPAYMENT = "overpayment"
+    COMMERCIAL_GESTURE = "commercial gesture"
+    OFFER_PRICE_REGULATION = "offer price regulation"
+    FRAUD = "fraud"
+
+
+class FinanceIncident(Base, Model):
+    id: int = sqla.Column(sqla.BigInteger, primary_key=True, autoincrement=True)
+    kind: IncidentType = sqla.Column(
+        sqla.Enum(IncidentType, native_enum=False, create_contraint=False),
+        nullable=False,
+    )
+    status: IncidentStatus = sqla.Column(
+        sqla.Enum(IncidentStatus, native_enum=False, create_constraint=False),
+        nullable=False,
+        server_default=IncidentStatus.CREATED.value,
+        default=IncidentStatus.CREATED,
+    )
+
+    venueId = sqla.Column(sqla.BigInteger, sqla.ForeignKey("venue.id"), nullable=False)
+    venue: sqla_orm.Mapped["offerers_models.Venue | None"] = sqla_orm.relationship(
+        "Venue", foreign_keys=[venueId], backref="finance_incidents"
+    )
+
+    details: dict | None = sqla.Column(
+        sqla_mutable.MutableDict.as_mutable(sqla_psql.JSONB), nullable=False, default={}, server_default="{}"
+    )
+
+    @property
+    def invoiced_pricing(self) -> Pricing | None:
+        processed_pricings = [pricing for pricing in self.pricings if pricing.status == PricingStatus.INVOICED]
+
+        if not processed_pricings:
+            return None
+
+        pricings = sorted(processed_pricings, key=attrgetter("creationDate"), reverse=True)
+
+        if pricings:
+            return pricings[0]
+
+        return None
```

Comment:
Comme l'indiquait Patrick, cette condition n'a pas de sens. On a déjà testé plus haut que `invoiced_pricings` n'était pas vide.  Donc ici on peut simplement écrire : 

```suggestion
        return sorted(processed_pricings, key=attrgetter("creationDate"), reverse=True)[0]
```